### PR TITLE
Srt support

### PR DESCRIPTION
**Pull request by adbrucker**
Hi,
I implemented rudimentary support for transcripts in SubRip (SRT) format. Please have a look. For me, it makes the OBS Caption Plugin much more useful for local recordings, as the SRT format allows for easy post-editing (correcting subtitles) and using tools such as Handbreak allow to properly add the subtitles to various video formats in such a way that many players support them. 

Best,
Achim

PS: The "heuristic" for the the start time for captions is rather ad hoc but seems to work "well enough" in my tests - for sure, there is a lot of room for improvement.

### CODE REVIEW

**Comment by ratwithacompiler** [comment]
Hi, so this is awkward but I'm gonna have to deny that pull request sorry.

So I kinda implemented a half working version of srt files a while ago like that but then on testing found it pretty weird and not great because whenever there's even just a short pause in speaking the speech API ends up splitting everything into it's own result so there were a ton short 1 or 2 second srt segments with only 2 or 3 words each which looked pretty bad to me. I started on a version that batched up multiple short sentences like that together into one if they were close enough timing wise to fix that but then got sidetracked and forgot about it tbh. 

The other issue with transcripts was that they were kinda broken in general and missing what was said at the end of an API session (Google's API has a 5 min max limit so you basically need to continually make new connections) because the way I handled those interruptions was kinda ugly and I really needed to clean that up first for transcripts to become usable and reliable.

So long story short your PR did finally kick me in the butt enough to get around to doing both of those things and I committed that now, with not missing results anymore on interruptions and srt support with UI to set a seconds value for batching and for splitting into multiple lines which hopefully should work reasonably okayish.

So I can't really merge that and apologize for the inconvenience but thank you very much for the PR and that is added now!
